src/test/java/seedu/address/logic/commands/VisitorInCommandTest.java

```diff
@@ -0,0 +1,224 @@
+        Visitor validPersonRecord = new Visitor("JACK");
+        new VisitorInCommand(null, validPersonRecord);
+    }
+
+    @Test
+    public void execute_patientVisitorAcceptedByModel_addSuccessful() throws Exception {
+        Person validPerson = new PersonBuilder().build();
+        Name validPersonName = validPerson.getName();
+        Visitor validPersonRecord = new Visitor("JACK");
+
+        ModelStubWithRegisteredPatient modelStub = new ModelStubWithRegisteredPatient(validPerson);
+        CommandResult commandResult = new VisitorInCommand(validPersonName, validPersonRecord)
+                .execute(modelStub, commandHistory);
+
+        assertEquals(String.format(VisitorInCommand.MESSAGE_SUCCESS, validPersonName), commandResult.feedbackToUser);
+        assertEquals(EMPTY_COMMAND_HISTORY, commandHistory);
+        //TODO assertEquals(DS storing strings, the DS in the model used to store)
+    }
+
+    @Test
+    public void execute_unregisteredPatient_throwsCommandException() throws Exception {
+        Person validPerson = new PersonBuilder().build();
+        ModelStub modelStub = new ModelStubWithRegisteredPatient(validPerson);
+
+        Person diffValidPerson = new PersonBuilder().withName("JACK").build();
+        Visitor visitor = new Visitor("Daniel");
+        VisitorInCommand visitorInCommand = new VisitorInCommand(diffValidPerson.getName(), visitor);
+
+        thrown.expect(CommandException.class);
+        thrown.expectMessage(VisitorInCommand.MESSAGE_UNREGISTERED);
+        visitorInCommand.execute(modelStub, commandHistory);
+    }
+
+    @Test
+    public void equals() {
+        Person jack = new PersonBuilder().withName("jack").build();
+        Person daniel = new PersonBuilder().withName("daniel").build();
+        Visitor jackVisitor = new Visitor("GAO JIAXIN");
+        Visitor danielVisitor = new Visitor("ENOCH");
+        VisitorInCommand addJackVisitorCommand = new VisitorInCommand(jack.getName(), jackVisitor);
+        VisitorInCommand addDanielVisitorCommand = new VisitorInCommand(jack.getName(), danielVisitor);
+
+        // same object -> returns true
+        assertTrue(addJackVisitorCommand.equals(addJackVisitorCommand));
+
+        // same values -> returns true
+        VisitorInCommand addJackVisitorCommandCopy = new VisitorInCommand(jack.getName(), jackVisitor);
+        assertTrue(addJackVisitorCommand.equals(addJackVisitorCommandCopy));
+
+        // different types -> returns false
+        assertFalse(addJackVisitorCommand.equals(0));
+
+        // null -> returns false
+        assertFalse(addJackVisitorCommand.equals(null));
+
+        // different person -> returns false
+        assertFalse(addJackVisitorCommand.equals(addDanielVisitorCommand));
+
+    }
+
+    /**
+     * A default model stub that have all of the methods failing.
+     */
+    private class ModelStub implements Model {
```

Comment:
This class has already been extracted into CommandTestUtil in a previous commit; did you pull the changes from the master branch before committing this?